### DISCUSSION TITLE
switch webpack to development mode when use "--debug"

### DISCUSSION THREAD

**Discussion by huww98** [vuepress/core#530] [Q&A] [answered]
Hi all, today I spent hours trying to debug the client-side error "Hydration completed but contains mismatches". To view the warnings from vue, I ended up adding the following to my config:

```javascript
  bundlerConfig: {
    configureWebpack(config, isServer, isBuild) {
      return { mode: 'development' };
    }
  },
```

How about we can simplify this by enabling the development mode by default when invoked as `yarn build --debug`

BTW, the mismatched element is the "hash" field from `@vuepress/plugin-debug`, maybe that component should be ClientOnly to avoid the error?

**Comment by Mister-Hope** [comment, marked as answer]
Track #734
